File: internal/types/localization.go
package types //nolint: revive,nolintlint

type Point struct {
	Coordinates struct {
		Lon float64 `json:"lon"`
		Lat float64 `json:"lat"`
	} `json:"coordinates"`
}

type LocalizationCorner struct {
	UpperLeft  Point `json:"upper-left"`
	UpperRight Point `json:"upper-right"`
	LowerLeft  Point `json:"lower-left"`
	LowerRight Point `json:"lower-right"`
}

type Localization struct {
	CachedObject

	Corner LocalizationCorner `json:"corner"`
}

File: frontend/src/models/localization.ts
import { type CachedObject } from "@/models/common";

export class Point {
  coordinates: {
    lon: number;
    lat: number;
  };

  constructor(coordinates: { lon: number; lat: number }) {
    this.coordinates = coordinates;
  }
}

export class LocalizationCorner {
  "upper-left": Point;
  "upper-right": Point;
  "lower-left": Point;
  "lower-right": Point;

  constructor(upper_left: Point, upper_right: Point, lower_left: Point, lower_right: Point) {
    this["upper-left"] = upper_left;
    this["upper-right"] = upper_right;
    this["lower-left"] = lower_left;
    this["lower-right"] = lower_right;
  }
}

export class Localization {
  corner: LocalizationCorner;
  cachedObject: CachedObject;

  constructor(corner: LocalizationCorner, cachedObject: CachedObject) {
    this.corner = corner;
    this.cachedObject = cachedObject;
  }
}
